extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-widget.tsx

```diff
@@ -0,0 +1,199 @@
+    }
+
+    protected updatePlugins = async () => {
+        console.log('> UPDATE plugins...');
+
+        this.ready = false;
+        this.update();
+
+        // await this.sleep(10000);
+        await this.fetchPlugins();
+    }
+
+    protected renderPluginList(): React.ReactNode {
+        const theList: React.ReactNode[] = [];
+        this.plugins.forEach(plugin => {
+            const container = this.renderPlugin(plugin);
+            theList.push(container);
+        });
+
+        return <div className='che-plugin-list'>
+            {theList}
+        </div>;
+    }
+
+    private renderPlugin(plugin: ChePluginMetadata) {
+        // return <div key={plugin.id} className='che-plugin'>
+        return <div key={plugin.id} className={this.pluginClassName(plugin)}>
+            <div className='che-plugin-icon'>
+                <img src={plugin.icon}></img>
+            </div>
+            <div className='che-plugin-info'>
+                <div className='che-plugin-title'>
+                    <div className='che-plugin-name'>{plugin.name}</div>
+                    <div className='che-plugin-version'>{plugin.version}</div>
+                </div>
+                <div className='che-plugin-state'>Installed</div>
+                <div className='che-plugin-description'>
+                    <div>
+                        <div>{plugin.description}</div>
+                    </div>
+                </div>
+                <div className='che-plugin-publisher'>{plugin.publisher}</div>
+                <div className='che-plugin-add'>Install</div>
+            </div>
+        </div>;
+
+        // return <div key={plugin.id} className={this.createPluginClassName(plugin)}>
+        //     <div className='column flexcontainer pluginInformationContainer'>
+        //         <div className='row flexcontainer'>
+        //             <div className='fa fa-puzzle-piece fa-2x fa-fw'></div>
+        //             <div title={plugin.name} className='pluginName noWrapInfo'>{plugin.name}</div>
+        //         </div>
+        //         <div className='row flexcontainer'>
+        //             <div className='pluginVersion'>{plugin.version}</div>
+        //         </div>
+        //         <div className='row flexcontainer'>
+        //             <div className='pluginDescription noWrapInfo'>{plugin.description}</div>
+        //         </div>
+        //         <div className='row flexcontainer'>
+        //             <div className='pluginPublisher noWrapInfo flexcontainer'>Vetal</div>
+        //         </div>
+        //     </div>
+        // </div>;
+
+    }
+
+    protected pluginClassName(plugin: ChePluginMetadata): string {
+        const classNames = ['che-plugin'];
+        if (this.isPluginInstalled(plugin)) {
+            classNames.push('che-plugin-installed');
+        }
+        return classNames.join(' ');
+    }
+
+    private installedPlugins: string[] = [
+        'org.eclipse.che.editor.theia:1.0.0',
```

Comment:
I think we should get this information from workspace config attributes.

Reply:
Yep, it's only draft for now